test/development/error-overlay/index.test.tsx

```diff
@@ -0,0 +1,31 @@
+import { nextTestSetup } from 'e2e-utils'
+
+describe('DevErrorOverlay', () => {
+  const { next } = nextTestSetup({
+    files: __dirname,
+  })
+
+  it('can get error code from RSC error thrown by framework', async () => {
+    await next.render('/known-rsc-error')
+    const browser = await next.browser('/known-rsc-error')
+    const errorCode = await browser.waitForElementByCss(
+      '[data-nextjs-error-code]'
+    )
```

Comment:
We'll eventually upstream these into [`toDisplayRedbox`](https://github.com/vercel/next.js/pull/73621) and then we can remove these specific tests.